### PR TITLE
test: split unified AI inference suite

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary

Split the 8,593-line `UnifiedAiInferenceRepository` test monolith into responsibility-owned suites for:

- prompt selection and repository-shell behavior
- multimodal inference execution
- response creation and post-processing
- concurrent entity changes and streamed tool-call accumulation
- normal tool dispatch and checklist automation
- tool validation and recovery paths

Shared deterministic fixtures, stream builders, mock wiring, GetIt scopes, and temp-directory cleanup now live in a non-executable helper library.

The split preserves all 115 conventional test declarations and the Glados property test. The largest executable is now 2,368 lines.

This adds no CI checks or guardrails and changes no production behavior.

## Verification

- all six resulting executable test files pass locally (116 tests including the Glados test)
- `fvm dart format .`: 0 files changed on the final pass
- `fvm flutter analyze`: no issues found
- `git diff --check`

## Summary by CodeRabbit

- **Tests**
  - Expanded AI inference coverage across execution, post-processing, concurrency, tool handling, and edge cases.
  - Added validation for streaming, multimodal inference, persistence, checklist operations, failures, and concurrent updates.
  - Improved JSON extraction coverage, including nested objects and escaped characters.

- **Refactor**
  - Consolidated shared test setup, mocks, fixtures, and utilities into a reusable harness.

- **Documentation**
  - Documented the responsibility-based test suite structure and decomposition progress.

### CODE REVIEW

**Comment by matthiasn** [comment]
@coderabbitai review

File: test/features/ai/repository/unified_ai_inference_repository_post_processing_test.dart [hunk 1]
```diff
@@ -0,0 +1,1222 @@
...
+        // Should complete without rethrowing
+        final statusChanges = <InferenceStatus>[];
+        await repository!.runInference(
+          entityId: 'test-id',
+          promptConfig: promptConfig,
+          onProgress: (_) {},
+          onStatusChange: statusChanges.add,
+        );
+
+        expect(statusChanges, [InferenceStatus.running, InferenceStatus.idle]);
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/repository/unified_ai_inference_repository_post_processing_test.dart:1115]: Fixed in cba00ecad. Both image and audio failure tests now retain the running→idle assertion and verify `updateJournalEntity` was attempted exactly once.

File: test/features/ai/repository/unified_ai_inference_repository_test_helpers.dart [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+  when(
+    () => mockAiConfigRepo.getConfigById(provider.id),
+  ).thenAnswer((_) async => provider);
+  when(
+    () => mockAiInputRepo.buildTaskDetailsJson(id: id),
+  ).thenAnswer((_) async => taskDetailsJson);
+}
+
+/// Stubs `AiInputRepository.createAiResponseEntry` to return null.
+void stubCreateAiResponseEntry(MockAiInputRepository mock) {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/repository/unified_ai_inference_repository_test_helpers.dart]: Fixed in cba00ecad. The helper documentation now describes the returned `AiResponseEntry` and the captured fields used to construct it.

File: test/features/ai/repository/unified_ai_inference_repository_tool_edges_test.dart [hunk 3]
```diff
@@ -0,0 +1,1126 @@
...
+              arguments:
+                  '{"items":[{"title":"Do something","isChecked":false}]}',
+            ),
+          ],
+          task: task,
+        );
+
+        expect(result, isFalse);
+      },
+    );
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/repository/unified_ai_inference_repository_tool_edges_test.dart:676]: Fixed in cba00ecad. The disappearing-task path now verifies checklist creation, the task refetch, and that no item is added; the related failure/invalid/unknown-tool tests now assert their expected interactions or absence of side effects.

File: test/features/ai/repository/unified_ai_inference_repository_tools_test.dart [hunk 4]
```diff
@@ -0,0 +1,1435 @@
...
+                DateTime(2024, 3, 15, 10, 30).millisecondsSinceEpoch ~/ 1000,
+          ),
+        );
+
+      await streamController.close();
+      await inferenceFuture;
+
+      // Verify that tool calls were processed
+      // In real implementation, this would have processed the add_multiple_checklist_items tool call
+    });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/repository/unified_ai_inference_repository_tools_test.dart:1340]: Fixed in cba00ecad. Both fragmented no-ID tool-call tests now assemble valid checklist payloads, verify dispatch to the auto-checklist service, and assert the captured item titles.

**Comment by matthiasn** [comment]
Review follow-up pushed in cba00ecad. I addressed all four inline findings and 15 of the 16 nitpicks: stronger interaction assertions, valid fragmented tool payloads, shared fake/fallback registration, non-null harness state, guaranteed GetIt scope cleanup, provider-factory consolidation, dead alias removal, accurate docs/names, the correct `Todos` stub, and a behavioral test for the injected auto-checklist service.

I intentionally did not turn the stream stubs into replayable/factory-backed streams. Each stub is scoped to one test and invoked once; the controller-backed concurrency tests deliberately model one live single-subscription stream. Replaying or buffering those streams would change the timing semantics being tested. If a future test needs two inference invocations from one stub, it should supply two explicit streams or introduce a factory at that call site.

Local verification: all six affected suites pass together (116 tests, including the Glados property case), full analyzer reports no issues, full repository format is clean.
